pkgs/os-specific/linux/kernel/linux-testing-bcachefs.nix

```diff
@@ -1,24 +1,36 @@
-{ lib, buildPackages, fetchFromGitHub, fetchpatch, perl, buildLinux, ... } @ args:
+{ lib
+, fetchpatch
+, kernel
+, date ? "2021-05-20"
+, commit ? "ac91176cb1f48e218d19026977ebe52078297c32"
+, diffHash ? "1flkvq7v35ay6x9z9519bylah8l1a787s95gq6bn30izwzq1vx6p"
+, kernelPatches ? [] # will always be supplied by callPackage because it is also a top-level attribute
+, argsOverride ? {}
+, ...
+} @ args:
 
-buildLinux (args // {
-  version = "5.9.0-2020.11.20";
-  modDirVersion = "5.9.0";
+kernel.override ( args // {
 
-  src = fetchFromGitHub {
-    owner = "koverstreet";
-    repo = "bcachefs";
-    # commit does not exist on any branch on the target repository
-    rev = "6a505b63ed3003faf5000f19fd08bbd477d93fbc";
-    sha256 = "1rf34gzv9npafp1c3i6lymk3b0gnqp4rb0wl33pw6yrpgnsry3cc";
-  };
+  argsOverride = {
+    version = "${kernel.version}-${date}";
+    extraMeta = {
+      branch = "master";
+      hydraPlatforms = []; # Should the testing kernels ever be built on Hydra?
+      maintainers = with lib.maintainers; [ davidak chiiruno ];
+      platforms = [ "x86_64-linux" ];
+    };
+  } // argsOverride;
 
-  extraConfig = "BCACHEFS_FS m";
+  kernelPatches = [
+    {
+      name = "bcachefs-${commit}";
+      patch = fetchpatch {
+        name = "bcachefs-${commit}.diff";
+        url = "https://evilpiepirate.org/git/bcachefs.git/rawdiff/?id=${commit}&id2=v${lib.versions.major kernel.version}.${lib.versions.minor kernel.version}";
```

Comment:
```suggestion
        url = "https://evilpiepirate.org/git/bcachefs.git/rawdiff/?id=${commit}&id2=v${lib.versions.majorMinor kernel.version}";
```

Reply:
Done, great suggestion. Sorry about the late reaction, I missed the mail. :/